packages/tocco-ui/src/Button/Button.js

```diff
@@ -2,8 +2,10 @@ import PropTypes from 'prop-types'
 import React from 'react'
```

Comment:
I agree that complexity of conditional styles is too high and component is too much entangled with other component. I have created story https://www.pivotaltracker.com/story/show/157047283 to tackle those deficits.

Reply:
I will tackle this issue if I know all components in package tocco-ui which rely on Buttons and after I have collected additional experience. A recommended strategy for styled-components is to cram all related styles into one file. What ensures an overview about all related declaration. 

Probably it will make sense to move/refactor setButtonDensity() and setButtonColor() as global utilStyles, likewise getElevation(). May there is also a possibility to rearrange styles in a smarter way to reduce resets and complexity.

Reply:
Several recent changes affect this issue. Current state should be satisfying.